checkers/assigning_to_self_checker.py

```diff
@@ -0,0 +1,32 @@
+"""checker for assigning to self.
+"""
+
+import astroid
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages
+
+
+class AssigningToSelfChecker(BaseChecker):
+
+    __implements__ = IAstroidChecker
+
+    name = 'always_return'
+    msgs = {'E9992': ('Assigning value to self on line %s',
+                      'assigning_to_self',
+                      'Used when you assign a value to self'),
+           }
+
+    # this is important so that your checker is executed before others
+    priority = -1
+
+    @check_messages("assigning_to_self")
+    def visit_assign(self, node):
+        target = node.targets[0]
+        if isinstance(target, astroid.AssignName) and target.as_string() == "self":
```

Comment:
Please check the attributes of `target` to see if you can directly access the name being assigned to, rather than converting the node into a string with `as_string`